apps/ejabberd/src/ejabberd_auth.erl

```diff
@@ -227,6 +227,8 @@ check_digest(Digest, DigestGen, _Password, Passwd) ->
                    Server :: ejabberd:server(),
                    Password :: binary()
                   ) -> ok | {error, empty_password | not_allowed | invalid_jid}.
+set_password(_,_, <<"">>) ->
+    {error, empty_password};
 set_password(_User, _Server, "") ->
```

Comment:
I'd remove the head with empty string `""` as it's never called (and shouldn't be).